e2e/single-cluster/agent_scheduling_customization_test.go

```diff
@@ -0,0 +1,247 @@
+package singlecluster_test
+
+import (
+	"context"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ptr[T any](v T) *T {
+	return &v
+}
+
+var _ = Describe("Agent Scheduling Customization", func() {
+	var (
+		cluster *fleet.Cluster
+	)
+
+	BeforeEach(func() {
+		var err error
+		cluster = &fleet.Cluster{}
+		err = clientUpstream.Get(context.TODO(), client.ObjectKey{
+			Namespace: env.Namespace,
+			Name:      "local",
+		}, cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	When("agentSchedulingCustomization.PriorityClass is configured on the cluster resource", func() {
+		BeforeEach(func() {
+			// Update the cluster with agentSchedulingCustomization
+			Eventually(func(g Gomega) {
+				// Re-fetch the cluster to get the latest version
+				latestCluster := &fleet.Cluster{}
+				err := clientUpstream.Get(context.TODO(), client.ObjectKey{
+					Namespace: env.Namespace,
+					Name:      "local",
+				}, latestCluster)
+				g.Expect(err).ToNot(HaveOccurred())
+
+				latestCluster.Spec.AgentSchedulingCustomization = &fleet.AgentSchedulingCustomization{
+					PriorityClass: &fleet.PriorityClassSpec{
+						Value:            1000,
+						PreemptionPolicy: ptr(corev1.PreemptNever),
+					},
+				}
+
+				err = clientUpstream.Update(context.TODO(), latestCluster)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).Should(Succeed(), "Should be able to update cluster with agentSchedulingCustomization")
+		})
+
+		AfterEach(func() {
+			// Clean up by removing the agentSchedulingCustomization
+			Eventually(func(g Gomega) {
+				// Re-fetch the cluster to get the latest version
+				latestCluster := &fleet.Cluster{}
+				err := clientUpstream.Get(context.TODO(), client.ObjectKey{
+					Namespace: env.Namespace,
+					Name:      "local",
+				}, latestCluster)
+				g.Expect(err).ToNot(HaveOccurred())
+
+				latestCluster.Spec.AgentSchedulingCustomization = nil
+				err = clientUpstream.Update(context.TODO(), latestCluster)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).Should(Succeed(), "Should be able to update cluster to remove agentSchedulingCustomization")
+
+			// Wait for PriorityClass to be deleted
+			Eventually(func() bool {
+				pc := &schedulingv1.PriorityClass{}
+				err := clientUpstream.Get(context.TODO(), client.ObjectKey{
+					Name: "fleet-agent-priority-class",
+				}, pc)
+				return client.IgnoreNotFound(err) == nil
```

Comment:
This will pass even if the priority class is not deleted, because `client.IgnoreNotFound(nil)` is `nil` ;)